docs/development.md

```diff
@@ -0,0 +1,46 @@
+# Development
+
+## Prerequisites
+
+You must have a recent version of [bazel](https://bazel.io) installed. Bazel is
+the recommended way to build and test the cluster registry. Bazel is designed to
+maintain compatibility with standard Go tooling, but this is not tested on a
+regular basis, and some scripts/tooling in this project are built around Bazel.
+
+NOTE: There is an issue with bazel 0.6.x. As a workaround, use 0.5.x, or pass
+the flag `--incompatible_comprehension_variables_do_not_leak=false` to bazel
+0.6.x invocations.
+
+Before doing any development work, you must (in order, from the repository root
+directory, after cloning):
+
+1.  run `./update-codegen.sh`
```

Comment:
Might want to point out that you need to clone this to

```
$GOPATH/src/k8s.io/cluster-registry
```

And not

```
$GOPATH/src/github.com/kubernetes/cluster-registry
```

Also kubernetes/kubernetes commits these artifacts so non-bazel projects can import the APIs. Any thoughts there?

Reply:
Good point. Though, interestingly enough, after #

Hmm. Personally I'd rather avoid committing artifacts until there are enough people asking for it.

Reply:
> Hmm. Personally I'd rather avoid committing artifacts until there are enough people asking for it.

works for me

Reply:
This script fails with:
```
generating clientset for group clusterregistry and version v1alpha1 at /home/ifont/go/src/k8s.io/cluster-registry/pkg/client
.............
INFO: Found 1 target...
Target @io_k8s_code_generator//cmd/client-gen:client-gen up-to-date:
  bazel-bin/external/io_k8s_code_generator/cmd/client-gen/client-gen
INFO: Elapsed time: 3.839s, Critical Path: 0.05s

INFO: Running command line: bazel-bin/external/io_k8s_code_generator/cmd/client-gen/client-gen --input-base k8s.io/cluster-registry/pkg/apis --input clusterregistry/ clusterregistry/v1alpha1 --clientset-path k8s.io/cluster-registry/pkg/client/clientset_generated '--output-base=/home/ifont/go/src'
ERROR: logging before flag.Parse: F1017 11:35:35.977724   16879 main.go:202] Error: Failed making a parser: unable to add directory "k8s.io/apimachinery/pkg/fields": unable to import "k8s.io/apimachinery/pkg/fields": cannot find package "k8s.io/apimachinery/pkg/fields" in any of:
        /usr/local/go/src/k8s.io/apimachinery/pkg/fields (from $GOROOT)
        /home/ifont/go/src/k8s.io/apimachinery/pkg/fields (from $GOPATH)
goroutine 1 [running]:
github.com/golang/glog.stacks(0xc42000e000, 0xc420118000, 0x17a, 0x1ce)
        external/com_github_golang_glog/glog.go:769 +0xa7
github.com/golang/glog.(*loggingT).output(0x8635c0, 0xc400000003, 0xc4200871e0, 0x84115f, 0x7, 0xca, 0x0)
        external/com_github_golang_glog/glog.go:720 +0x36b
github.com/golang/glog.(*loggingT).printf(0x8635c0, 0xc400000003, 0x6ffc79, 0x9, 0xc4200e5e70, 0x1, 0x1)
        external/com_github_golang_glog/glog.go:655 +0x14f
github.com/golang/glog.Fatalf(0x6ffc79, 0x9, 0xc4200e5e70, 0x1, 0x1)
        external/com_github_golang_glog/glog.go:1148 +0x67
main.main()
        external/io_k8s_code_generator/cmd/client-gen/main.go:202 +0x6c5
ERROR: Non-zero return code '255' from command: Process exited with status 255.
```
Unless you `go get k8s.io/apimachinery` first as a workaround. We appear to be missing this dependency.

Reply:
@font I think you're hitting the issue https://github.com/kubernetes/cluster-registry/pull/18 addresses.